doc/guides/using-internal-errors.md

```diff
@@ -0,0 +1,136 @@
+# Using the internal/errors.js Module
+
+## What is internal/errors.js
+
+The `require('internal/errors')` module is an internal-only module that can be
+used to produce `Error`, `TypeError` and `RangeError` instances that use a
+static, permanent error code and an optionally parameterized message. The
+intent of the module is to make errors thrown by Node.js core to be more
+consistent and to allow changes to error messages to be made without requiring
+a semver-major label.
```

Comment:
Nit: `semver-major label` is kind of our internal jargon and people aren't going to understand (without an explanation) why we consider error message changes as semver major anyway.

Maybe just embrace completeness as this is a guide and go with something more like this?:

> ...parameterized message. [END OF PARAGRAPH]
>
> The intent of the module is to allow errors provided by Node.js to be assigned a 
> permanent identifier. Without a permanent identifier, userland code may need to inspect
> error messages to distinguish one error from another. An unfortunate result
> of that practice is that changes to error messages result in broken code in the ecosystem.
> For that reason, Node.js has considered error message changes to be breaking changes.
> By providing a permanent identifier for a specific error, we reduce the need for userland
> code to inspect error messages.